cpp/log/etcd_consistent_store-inl.h

```diff
@@ -84,8 +148,13 @@ template <class Logged>
 EtcdConsistentStore<Logged>::~EtcdConsistentStore() {
   VLOG(1) << "Cancelling watch tasks.";
   serving_sth_watch_task_.Cancel();
+  cluster_config_watch_task_.Cancel();
   VLOG(1) << "Waiting for watch tasks to return.";
   serving_sth_watch_task_.Wait();
+  cluster_config_watch_task_.Wait();
+  VLOG(1) << "Cancelling stats task.";
+  etcd_stats_task_.Cancel();
```

Comment:
Would be slightly better to cancel everything, then wait on everything?

This has a slight smell that maybe we should just have one SyncTask, and use child tasks for all these things, but don't do this in this PR, just putting the idea in your head... :wink: